File: scripts/prebuild-site.js
#!/usr/bin/env node
/**
 * prebuild.js — gera catalog.json para o site estatico
 * Copia catalog.yaml como JSON para web/public/
 */
const fs = require("fs");
const path = require("path");
const yaml = require("js-yaml");

const root = path.resolve(__dirname, "..");
const catalogYaml = path.join(root, "knowledge-factory", "registry", "catalog.yaml");
const catalogJson = path.join(root, "web", "public", "catalog.json");

if (!fs.existsSync(catalogYaml)) {
  console.warn("catalog.yaml nao encontrado. Pulando geracao.");
  process.exit(0);
}

const doc = yaml.load(fs.readFileSync(catalogYaml, "utf8"));
// Simplify: keep only what the frontend needs
const simplified = {
  version: doc.version,
  items: doc.items.map((item) => ({
    id: item.id,
    type: item.type,
    title: item.title,
    taxonomy: item.taxonomy,
  })),
};

fs.writeFileSync(catalogJson, JSON.stringify(simplified, null, 2), "utf8");
console.log(`catalog.json gerado com ${simplified.items.length} items.`);

File: scripts/prebuild_site.py
#!/usr/bin/env python3
"""
prebuild_site.py
Gera web/public/catalog.json a partir de knowledge-factory/registry/catalog.yaml.
"""

import json
import sys
from pathlib import Path

import yaml

ROOT = Path(__file__).resolve().parent.parent
CATALOG_YAML = ROOT / "knowledge-factory" / "registry" / "catalog.yaml"
CATALOG_JSON = ROOT / "web" / "public" / "catalog.json"

if not CATALOG_YAML.exists():
    print("catalog.yaml nao encontrado. Pulando.", file=sys.stderr)
    sys.exit(0)

with open(CATALOG_YAML, encoding="utf-8") as f:
    doc = yaml.safe_load(f)

simplified = {
    "version": doc.get("version", 1),
    "items": [
        {
            "id": item["id"],
            "type": item.get("type", "unknown"),
            "title": item.get("title", ""),
            "taxonomy": item.get("taxonomy", {}),
        }
        for item in doc.get("items", [])
    ],
}

CATALOG_JSON.parent.mkdir(parents=True, exist_ok=True)
with open(CATALOG_JSON, "w", encoding="utf-8") as f:
    json.dump(simplified, f, indent=2, ensure_ascii=False)

print(f"catalog.json gerado com {len(simplified['items'])} items em {CATALOG_JSON}")
